From 942469249453b4e0e9dc07ae35d256df39de3fe7 Mon Sep 17 00:00:00 2001
From: Daniel Scally <djrscally@gmail.com>
Date: Wed, 19 Aug 2020 23:19:57 +0900
Subject: [PATCH 1/5] mfd: tps68470: add tps68470_i2c_ids for sysfs new_device

(from gist found in this comment
- https://github.com/linux-surface/linux-surface/issues/91#issuecomment-675884262)
[Added comment to above tps68470_i2c_ids. Added MODULE_DEVICE_TABLE(),
useful when driver is built as module.]
[Usage:
    $ echo "INT3472 0x0c" | sudo tee /sys/bus/i2c/devices/i2c-8/new_device]
Signed-off-by: Tsuchiya Yuto <kitakar@gmail.com>
---
 drivers/mfd/tps68470.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mfd/tps68470.c b/drivers/mfd/tps68470.c
index 853adef4f920..5a2b0c32d160 100644
--- a/drivers/mfd/tps68470.c
+++ b/drivers/mfd/tps68470.c
@@ -93,12 +93,20 @@ static const struct acpi_device_id tps68470_acpi_ids[] = {
 };
 MODULE_DEVICE_TABLE(acpi, tps68470_acpi_ids);
 
+/* To be used with sysfs new_device */
+static const struct i2c_device_id tps68470_i2c_ids[] = {
+	{"INT3472", 0},
+	{},
+};
+MODULE_DEVICE_TABLE(i2c, tps68470_i2c_ids);
+
 static struct i2c_driver tps68470_driver = {
 	.driver = {
 		   .name = "tps68470",
 		   .acpi_match_table = tps68470_acpi_ids,
 	},
 	.probe_new = tps68470_probe,
+	.id_table = tps68470_i2c_ids,
 	.remove = tps68470_remove,
 };
 module_i2c_driver(tps68470_driver);
-- 
2.28.0

